actions/docxUpdater.js

```diff
@@ -0,0 +1,142 @@
+ * @param {Array} mdast - The mdast to be updated.
+ * @param {string} expName - The name of the experience.
+ * @param {RegExp} grayboxStylePattern - The pattern to match graybox styles.
+ */
+const updateExperienceNameFromLinks = (mdast, expName) => {
+    if (mdast) {
+        mdast.forEach((child) => {
+                if (child.type === 'gridTable') {
+                    firstGtRows.push(findFirstGtRowInNode(child));
+                }
+                //remove experience name from links on the document
+                if (child.type === 'link' && child.url && (child.url.includes(expName) || child.url.includes(gbDomainSuffix))) {
+                    child.url = child.url.replaceAll(`/${expName}/`, '/').replaceAll(gbDomainSuffix, emptyString);
+                    logger.info(`Link updated: ${child.url}`);
+                }
+                if (child.children) {
+                    updateExperienceNameFromLinks(child.children, expName);
+                }
+            }
+        );
+    }
+}
+
+/**
+ * Helper function, iterates through the firstGtRows array and replaces graybox styles for each row.
+ */
+const iterateGtRowsToReplaceStyles = () => {
+    firstGtRows.forEach((gtRow) => {
+        if (gtRow && gtRow.children) {
+            replaceGrayboxStyles(gtRow);
+        }
+    });
+}
+
+/**
+ * Replaces all graybox styles from blocks and text.
+ * 
+ * @param {object} node - The node to process.
+ * @returns {void}
+ */
+const replaceGrayboxStyles = (node) => {
+    //replace all graybox styles from blocks and text
+    if (node && node.type === 'text' && node.value && node.value.includes(gbStyleExpression)) {
+        logger.info(node);
+        node.value = node.value.replace(grayboxStylesRegex, emptyString)
+            .replace('()', emptyString).replace(', )', ')');
+        logger.info('updated value>>  ');
+        logger.info(node);
+        return;
+    }
+    if (node.children) {
+        node.children.forEach((child) => {
+            replaceGrayboxStyles(child);
+        });
+    }
+}
+
+/**
+ * Finds the first 'gtRow' node in the given node or its children.
+ * @param {Object} node - The node to search in.
+ * @returns {Object|undefined} - The first 'gtRow' node found, or undefined if not found.
+ */
+function findFirstGtRowInNode(node) {
+    if (node && node.type === 'gtRow') {
+        return node;
+    }
+    if (node.children) {
+        for (const child of node.children) {
+            return findFirstGtRowInNode(child);
+        }
+    }
+}
+
+
+/**
+ * Generate a Docx file from the given mdast.
+ * @param {Object} mdast - The mdast representing the document.
+ * @returns {Promise} A promise that resolves to the generated Docx file.
+ */
+async function generateDocxFromMdast(mdast) {
+    logger.info('Docx file Docx file generation from mdast started...');
+    return await mdast2docx(mdast);   
```

Comment:
I am assuming this fails without the styles.xml passed in. Not sure.

Reply:
it worked in the standalone app but haven't tested in the integrated adobeio layer.